src/components/DateRangeInput.tsx

```diff
@@ -61,7 +61,7 @@ export default function DateRangeInput({
       onChange={(newValue) => {
         if (newValue) {
           onChange({
-            start: DateTime.fromISO(newValue.startDate as string),
+            start: DateTime.fromISO(newValue.startDate as string).endOf('day'),
```

Comment:
the actual fix! this makes Interval.splitBy to include the extra interval for when we select first day of the month